src/server/api/general.js

```diff
@@ -435,6 +435,36 @@ export default function useGeneralApi(app) {
         }
         this.body = JSON.stringify({status: 'ok'});
     });
+
+    router.post('/setUserPreferences', koaBody, function *() {
+        const params = this.request.body;
+        const {csrf, payload} = typeof(params) === 'string' ? JSON.parse(params) : params;
+        if (!checkCSRF(this, csrf)) return;
```

Comment:
send an error code and message here (403 maybe?)

Reply:
checkCSRF does exactly that - it sets 403 status if check fails

Reply:
Sorry, I meant the side effects should be here in this method as opposed to in `checkCSRF`.

Reply:
I prefer pure functions too, but we are using checkCSRF in many places, so moving 403 out of checkCSRF will require a lot work and changes, I think this should be done separately from this PR

Reply:
:+1: